src/main/java/net/sf/jabref/gui/worker/VersionWorker.java

```diff
@@ -0,0 +1,90 @@
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.gui.worker;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.swing.JOptionPane;
+import javax.swing.SwingWorker;
+
+import net.sf.jabref.gui.JabRefFrame;
+import net.sf.jabref.gui.help.NewVersionDialog;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.version.Version;
+import net.sf.jabref.logic.util.version.VersionChecker;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class VersionWorker extends SwingWorker<Void, Void> {
+
+    private static final Log LOGGER = LogFactory.getLog(VersionWorker.class);
+
+    private final JabRefFrame mainFrame;
+    private final boolean manualExecution;
+    private final Version currentVersion;
+    private final Version toBeIgnored;
+
+    public VersionWorker(JabRefFrame mainFrame, boolean manualExecution, Version currentVersion, Version toBeIgnored) {
+        this.mainFrame = Objects.requireNonNull(mainFrame);
+        this.manualExecution = manualExecution;
+        this.currentVersion = Objects.requireNonNull(currentVersion);
+        this.toBeIgnored = Objects.requireNonNull(toBeIgnored);
+    }
+
+    @Override
+    protected Void doInBackground() throws Exception {
+        Version latestVersion;
+        try {
+            latestVersion = VersionChecker.grabLatestVersion();
+        } catch (IOException ioException) {
+            String couldntConnect = Localization.lang("Couldn't connect to the update server");
+            String tryLater = Localization.lang("Please try again later and/or check your network connection");
+            if (manualExecution) {
+                JOptionPane.showMessageDialog(this.mainFrame,
+                        couldntConnect + "\n" + tryLater,
+                        couldntConnect,
+                        JOptionPane.ERROR_MESSAGE);
+            }
+            this.mainFrame.output(couldntConnect + " " + tryLater);
+            LOGGER.warn("Couldn't connect to the updateserver", ioException);
+            return null;
+        }
+
+        // only respect the ignored version on automated version checks
+        if (latestVersion.equals(toBeIgnored) && !manualExecution) {
```

Comment:
maybe even latestVersion <= toBeIgnored (which shouldn't be necessary in most of the cases).